src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,33 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    private static final int MIN_AGE_FOR_PRESIDENCY = 35;
+    private static final int MIN_YEARS_LIVED_IN_UKRAINE = 10;
+    private static final String REQUIRED_NATIONALITY = "Ukrainian";
+
+    @Override
+    public boolean test(Candidate candidate) {
+        Predicate<Candidate> agePredicate = c -> c.getAge() >= MIN_AGE_FOR_PRESIDENCY;
```

Comment:
```suggestion
        return candidate.getAge() >= MIN_AGE
                && candidate.isAllowedToVote()
                && candidate.getNationality().equals(REQUIRED_NATIONALITY)
                && checkTimeLivingInCountry(candidate);
```

create `private boolean checkTimeLivingInCountry(Candidate candidate)`